test/pytest/test_hotplug.py

```diff
@@ -0,0 +1,18 @@
+import serial
+import pytest
+from pexpect.fdpexpect import fdspawn
+
+@pytest.mark.timeout(400)
+
+def test_hotplug(qemu_instance_per_fcn):
+    qemu_instance_per_fcn['serial2'].sendline('root')
+    cpu_path = '/sys/devices/system/cpu/'
+
+    for i in range(1,8):
```

Comment:
Instead of having a loop within the test, it's preferable to have a separate test for each core.  That way, if the test for one of the cores fails, it's easier to see which one, and then the test will continue to the next core.  You can do this by using removing the for loop and placing something like "@pytest.mark.parametrize('core_num', range(1,8))" at the top of the test.  You'll also have to change the test signature.  Please see test_mbox.py for an example.